docs_src/design/adr/core/0022-UoM.md

```diff
@@ -0,0 +1,199 @@
+
+**Example A Metadata Configuration - external definition of UoM**
+``` toml
+[Writable]
+    [Writable.UoM]
+    validation=off  ## off by default
+
+## in the non-writable area - example external file specified to units of measure
+[UoM]
+    UoMFile="file::/tmp/edgex/metadata/uom.toml"  # the UoMFile location is specified as a URI.
+```
+
+The location of the external UoM file (specified by URI) should point to an accessible website, shared volume, volume mount, etc. in order to allow for the adopter to provide the units of measure independently during configuration/setup of the EdgeX instance without requiring a build of the metadata service or a reconstruction of the Docker image/container.
+
+!!! Info
+        In future versions, multiple external UoM definition files might be specified or use of inline and external files might be allowed simultaneously.  This may help the organization of the units in the future.  In the first iteration of this functionality, either one external file *or* inline definition is allowed.
+
+!!! Note
+    When the units of measures is provided by external UoM definition file, then *environmental overrides can be used to specify and override the location of the external configuration file*.
+
+#### Inline UoM definition
+When the units of measure are specified directly in metadata's configuration file (i.e. inline definition), the actual units of measure are provided in core metadata's configuration (currently in `res/configuration.toml`) - see example B below.  
+
+**Example B Metadata Configuration - inline definition of UoM**
+``` toml
+[Writable]
+    [Writable.UoM]
+    validation=false  ## off by default
+
+## in the non-writable area - example inline units of measure
+[Uom]
+    source="reference to source for all UoM if not specified below"
+    [Uom.temperature]
+    source="www.weather.com"
+    values="C,F,K"
+    [Uom.weights]
+    source="www.usa.gov/federal-agencies/weights-and-measures-division"
+    values="lbs,ounces,kilos,grams"
+```
+
+!!! Note
+    When the units of measure are defined in-line, then *environmental overrides can be used to specify and override the units of measure in the configuration.toml file*.  However, since the Uom is a map, the unit of measure category must already exist in order to be overridden.  In the example above, you could provide an environmental override to replace `Uom.weights` but you could not provide an override for `Uom.speed` as it does not exist in the original configuration file.
+
+#### Specifying Validation on or off
+Additionally, in either external or inline definition, a configuration option for unit of measure validation being `on` or `off` will be provided (note `validation` in both example A and B above).  The units of measure or the location of the UoM file is static, but the ability to turn validation on/off is dynamic and therefore in the writable area of configuration.  For backward compatibility, validation will be off by default.
```

Comment:
Is using on/off a project convention for boolean values?

Reply:
I added a note to state that on and off are represented by booleans true and false in configuration.